docs: Add Karpenter support for Windows docs (#4074)
aws/karpenter: website/content/en/preview/faq.md

### Does Karpenter support IPv6?

> Yes! Karpenter dynamically discovers if you are running in an IPv6 cluster by checking the kube-dns service's cluster-ip. When using an AMI Family such as `AL2`, Karpenter will automatically configure the EKS Bootstrap script for IPv6. Some EC2 instance types do not support IPv6 and the Amazon VPC CNI only supports instance types that run on the Nitro hypervisor. Additionally, Windows nodes do not support IPv6. It's best to add a requirement to your Provisioner to only allow Nitro instance types:

Reviewer: Can we make this a Windows Support Notice like the other ones throughout the docs?

For more documentation on enabling IPv6 with the Amazon VPC CNI, see the [docs](https://docs.aws.amazon.com/eks/latest/userguide/cni-ipv6.html).

{{% alert title="Windows Support Notice" color="warning" %}}

> Windows nodes do not support IPv6.

Reviewer 2: Same here

Reviewer: Tracking in a follow-on

### Can I set `--max-pods` on my nodes?
Not yet.

### Why do the Windows2019 and Windows2022 AMI families only support Windows Server Core?
The difference between the Core and Full variants is that Core is a minimal OS with less components and no graphic user interface (GUI) or desktop experience.
Windows AMI families use the Windows Server Core option for simplicity, but if required, you can specify a custom AMI to run Windows Server Full.

> Below is an example of how you can configure provisioning using the latest [Amazon EKS optimized AMI](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-windows-ami.html) with Windows Server 2022 Full for Kubernetes 1.27 by specify an `amiSelector`.

Reviewer: ```suggestion
You can specify the [Amazon EKS optimized AMI](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-windows-ami.html) with Windows Server 2022 Full for Kubernetes 1.27 by configuring an `amiSelector` that references the AMI name.
```

### Can I set `--max-pods` on my nodes?
Not yet.

### Why do the Windows2019 and Windows2022 AMI families only support Windows Server Core?
The difference between the Core and Full variants is that Core is a minimal OS with less components and no graphic user interface (GUI) or desktop experience.
`Windows2019` and `Windows2022` AMI families use the Windows Server Core option for simplicity, but if required, you can specify a custom AMI to run Windows Server Full.

> You can specify the [Amazon EKS optimized AMI](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-windows-ami.html) with Windows Server 2022 Full for Kubernetes 1.27 by configuring an `amiSelector` that references the AMI name.

Reviewer 3: It can be a follow-up. We should consider adding a tutorial somewhere with a sample NodeTemplate for launching instances based on Full AMIs.

Launching both Core and Full should be available as a step by step so that customers do not have to do substantial work/research on their end to configure Windows specific things.

Author: I considered the same and I was thinking this could be addressed in an AWS blog post but you are also right that we should have easy steps someone can read and follow in Karpenter docs itself. Lets address this as a follow up.
